scipy/optimize/tests/test_linprog.py

```diff
@@ -359,6 +358,48 @@ def f(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None):
 
         assert_raises(ValueError, f, [1, 2], A_ub=np.zeros((1, 1, 3)), b_eq=1)
 
+    def test_bounds_fixed(self):
+
+        def g(c, bounds):
+            res = linprog(c, bounds=bounds, method=self.method, options=self.options)
+            return res
+
+        # Test fixed bounds (upper equal to lower)
+        res = g([1], bounds=(1, 1))
+        _assert_success(res, 1, 1)
+        res = g([1, 2, 3], bounds=[(5, 5), (-1, -1), (3, 3)])
+        _assert_success(res, 12, [5, -1, 3])
+        res = g([1, 1], bounds=[(1, 1), (1, 3)])
+        _assert_success(res, 2, [1, 1])
+
+    def test_bounds_infeasible(self):
```

Comment:
Same as above; consider parameterizing and checking that the problem is solved in presolve.

Reply:
See above. Changed it so that the function is only used where errors are expected.

Reply:
OK I was also suggesting you parameterize these to make the test less repetitive:
https://docs.pytest.org/en/latest/parametrize.html
`c`, `bounds`, the optimal objective, and the solution would be parameters.

Reply:
If you address this and the (few) other new comments and if code coverage looks good, I think I can merge.